cmd/vela-makisu/build.go

```diff
@@ -4,15 +4,544 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	// check if any redis options were passed
+	if len(b.RedisCacheRaw) > 0 {
+		// cast raw http options into bytes
+		redisOpts := []byte(b.RedisCacheRaw)
+
+		// serialize raw http options into expected RedisCache type
+		err := json.Unmarshal(redisOpts, &b.RedisCache)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
 }
 
 // Validate verifies the Build is properly configured.
 func (b *Build) Validate() error {
 	logrus.Trace("validating build plugin configuration")
 
+	// verify tag are provided
+	if len(b.Context) == 0 {
+		return fmt.Errorf("no build context provided")
+	}
+
+	// verify tag are provided
+	if len(b.Tag) == 0 {
+		return fmt.Errorf("no build tag provided")
+	}
+
 	return nil
 }
+
+// Flags formats and outputs the flags for
+// configuring a Docker.
+func (d *Docker) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Host is provided
+	if len(d.Host) > 0 {
+		// add flag for Host from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-host=%s", d.Host))
+	}
+
+	// check if Scheme is provided
+	if len(d.Scheme) > 0 {
+		// add flag for Scheme from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-scheme=%s", d.Scheme))
+	}
+
+	// check if Version is provided
+	if len(d.Version) > 0 {
+		// add flag for Version from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-version=%s", d.Version))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a http cache.
+func (h *HTTPCache) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(h.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-addr=%s", h.Addr))
+	}
+
+	// check if Headers is provided
+	if len(h.Headers) > 0 {
+		var args string
+		for _, arg := range h.Headers {
+			args += fmt.Sprintf(" %s", arg)
+		}
+		// add flag for BuildArgs from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-header=%s", strings.TrimPrefix(args, " ")))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a redis cache.
+func (r *RedisCache) Flags() ([]string, error) {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(r.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-addr=%s", r.Addr))
+	}
+
+	// check if Password is provided
+	if len(r.Password) > 0 {
+		// add flag for Password from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-password=%s", r.Password))
+	}
+
+	// check if TTL is provided
+	if len(r.TTL) > 0 {
+		duration, err := time.ParseDuration(r.TTL)
+		if err != nil {
+			return nil, err
+		}
+
+		// check if TTL is valid duration
+		if len(duration.String()) != 0 {
```

Comment:
I still think this can be optimized further 👍 

You should be able to use the `duration` value without any extra checks.

```suggestion
		if duration > 0 {
```

Reply:
ahhh, I honestly did not understand what you meant when you said duration is of type `int64` because I was seeing it's type `time.Duration`

I see now looking at library code `time.Duration` is actually a custom type that is `int64`:
```// A Duration represents the elapsed time between two instants
// as an int64 nanosecond count. The representation limits the
// largest representable duration to approximately 290 years.
type Duration int64```


That was not clear in your previous messages which made me chose a different implementation again.

I can simplify it again.